### PR TITLE
Fixing "Too long for Unix domain socket" error deployphp/deployer#1356

### PR DESCRIPTION
**Pull request by nvaken**
| Q             | A
| ------------- | ---
| Bug fix?      | Yes
| New feature?  | No
| BC breaks?    | No
| Deprecations? | No
| Fixed tickets | #1356 

See #1356 for more info.

### CODE REVIEW

File: src/Ssh/Arguments.php [hunk 1]
```diff
@@ -127,31 +127,34 @@ public function withMultiplexing(Host $host)
      */
     private function generateControlPath(Host $host)
     {
+        $connectionHashLength = 16; // Length of connection hash that OpenSSH appends to controlpath
+        $unixMaxPath = 104; // Theoretical max limit for path length
+        $homeDir = getenv("HOME");
```
Comment (antonmedv) [inline src/Ssh/Arguments.php]: Please, use https://github.com/deployphp/deployer/blob/master/src/Support/Unix.php#L18

File: src/Ssh/Arguments.php [hunk 2]
```diff
@@ -127,31 +127,34 @@ public function withMultiplexing(Host $host)
...
                     break;
                 case 5:
                     throw new Exception("The multiplexing control path is too long. Control path is: $controlPath");
                 default:
-                    $controlPath = "~/.ssh/deployer_$connectionData";
+                    $controlPath = "$homeDir/.ssh/deployer_$connectionData";
             }
             $tryLongestPossible++;
-        } while (strlen($controlPath) > 104); // Unix socket max length
+        } while (strlen($controlPath) + $connectionHashLength > 104); // Unix socket max length
```
Comment (antonmedv) [inline src/Ssh/Arguments.php]: Nice. But use your $unixMaxPath var)

File: src/Ssh/Arguments.php (hunk 1 above)
Comment (nvaken) [reply src/Ssh/Arguments.php]: Thanks, didn't know about that one. Will fix.

File: src/Ssh/Arguments.php (hunk 2 above)
Comment (nvaken) [reply src/Ssh/Arguments.php]: Whoopsie daisies. ;-) Will do!

File: src/Ssh/Arguments.php [hunk 3]
```diff
@@ -127,31 +127,34 @@ public function withMultiplexing(Host $host)
...
-                    $controlPath = "~/deployer_$connectionData";
+                    $controlPath = "$homeDir/.ssh/deployer_$connectionData";
                     break;
                 case 3:
-                    $controlPath = "~/deployer_%C";
+                    $controlPath = "$homeDir/.ssh/deployer_%C";
                     break;
                 case 4:
-                    $controlPath = "~/mux_%C";
+                    $controlPath = "$homeDir/.ssh/mux_%C";
```
[the remark this answers is not in the document]
Comment (antonmedv) [reply src/Ssh/Arguments.php]: Yes, you are right.

[the remark this answers is not in the document]
Comment (nvaken) [reply src/Ssh/Arguments.php]: Great, I'll be submitting a new commit with these small fixes in a few.

**Comment by nvaken** [comment]
Fixed the small issues you've found @antonmedv, running through tests. Please review. 👍

**Pull request merged**